nodejs/eks/nodegroup.ts

```diff
@@ -1232,30 +1298,83 @@ function createNodeGroupV2Internal(
             },
             { parent },
         ),
-    ).blockDeviceMappings[0].deviceName;
+    );
 
-    const nodeLaunchTemplate = new aws.ec2.LaunchTemplate(
-        `${name}-launchTemplate`,
-        {
-            imageId: amiId,
-            instanceType: args.instanceType || "t2.medium",
-            iamInstanceProfile: { arn: instanceProfileArn },
-            keyName: keyName,
-            instanceMarketOptions: marketOptions,
-            blockDeviceMappings: [
+    const deviceName = pulumi.all([os, amiInfo]).apply(([os, amiInfo]) => {
+        if (os !== OperatingSystem.Bottlerocket) {
+            return amiInfo.blockDeviceMappings[0].deviceName;
+        }
+
+        // Bottlerocket has two block devices, the root device stores the OS itself and the other is for data like images, logs, persistent storage
+        // We need to allow users to configure the block device for data
+        const deviceName = amiInfo.blockDeviceMappings.find(
+            (bdm) => bdm.deviceName !== amiInfo.rootDeviceName,
+        )?.deviceName;
+        return deviceName ?? amiInfo.rootDeviceName;
+    });
+
+    const blockDeviceMappings = os.apply((os) => {
+        // Old behavior to stay backwards compatible
+        if (os === OperatingSystem.AL2) {
+            return [
                 {
-                    deviceName: device,
+                    deviceName: deviceName,
                     ebs: {
-                        encrypted: args.nodeRootVolumeEncrypted ?? false ? "true" : "false",
+                        encrypted: pulumi
+                            .output(args.nodeRootVolumeEncrypted)
+                            .apply((val) => (val === true ?? false ? "true" : "false")),
```

Comment:
```suggestion
                    .apply((val) => (val ? "true" : "false")),
```

I think this is what you mean?

Reply:
Ah yeah that's the same... good catch!